src/battle_main.c

```diff
@@ -299,27 +299,220 @@ const struct OamData gOamData_BattleSpritePlayerSide =
 
 static const s8 sCenterToCornerVecXs[8] ={-32, -16, -16, -32, -32};
 
-const u8 gTypeNames[NUMBER_OF_MON_TYPES][TYPE_NAME_LENGTH + 1] =
+#define GENERIC(type) .generic = _(#type" move")
+#define MOVES(z, max)               \
+        .zMove = MOVE_##z,          \
+        .maxMove = MOVE_MAX_##max,
+
+#define ITEMS(type, item, berryName, crystal) \
+        .enhanceItem = ITEM_##item,           \
+        .berry = ITEM_##berryName##_BERRY,    \
+        .gem = ITEM_##type##_GEM,             \
+        .zCrystal = ITEM_##crystal##IUM_Z,
+
+//Not in ITEMS since Normal doesn't have these two items (Blank Plate is not yet implemented)
+#define LEGENDARY_ITEMS(plateName, type)   \
+        .plate = ITEM_##plateName##_PLATE, \
+        .memory = ITEM_##type##_MEMORY,    \
+
+//Not in ITEMS since Stellar has Tera Shards but no other item
+#define TERA_SHARD(type) .teraShard = ITEM_##type##_TERA_SHARD,
```

Comment:
These macros are imo not worth the sacrifice of transparency. `FIST` does not clearly refer to a plate unless you know how the macro works and it will not show up in the results if you do a repo-wide search of `FIST_PLATE`.